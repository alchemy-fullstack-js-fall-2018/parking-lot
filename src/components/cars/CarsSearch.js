import React, { Fragment } from 'react';
import Cars from './Cars';

const CarsSearch = ({ cars, searchTerm, updateSearchTerm, totalCars }) => {
  return (
    <Fragment>
      <p>search page</p>
      <input value={searchTerm} onChange={updateSearchTerm} />
      <Cars cars={cars} totalCars={totalCars} />
    </Fragment>
  );
};

export default CarsSearch;
