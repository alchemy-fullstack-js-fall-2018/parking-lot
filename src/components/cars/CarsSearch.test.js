import React from 'react';
import { shallow } from 'enzyme';
import CarsSearch from './CarsSearch';
jest.mock('../../routes/routes.js');

describe('CarsSearch snapshot', () => {
  it('matches previous snapshot', () => {
    const cars = [
      {
        id: 'ABC123',
        make: 'Honda',
        model: 'Accord',
        color: 'white',
        state: 'OR'
      },
      {
        id: 'DEF456',
        make: 'Toyota',
        model: 'Camry',
        color: 'silver',
        state: 'OR'
      },
      {
        id: 'GHI789',
        make: 'Hyundai',
        model: 'Elantra',
        color: 'blue',
        state: 'WA'
      }
    ];

    const totalCars = 4;
    const searchTerm = 'ABC';
    const updateSearchTerm = jest.fn();

    const wrapper = shallow(
      <CarsSearch
        cars={cars}
        searchTerm={searchTerm}
        updateSearchTerm={updateSearchTerm}
        totalCars={totalCars}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
