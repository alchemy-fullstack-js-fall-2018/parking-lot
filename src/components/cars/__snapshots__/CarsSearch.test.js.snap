// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsSearch snapshot matches previous snapshot 1`] = `
<Fragment>
  <h2>
    Search for Car By Plate
  </h2>
  <input
    onChange={[MockFunction]}
    value="ABC"
  />
  <Cars
    cars={
      Array [
        Object {
          "color": "white",
          "id": "ABC123",
          "make": "Honda",
          "model": "Accord",
          "state": "OR",
        },
        Object {
          "color": "silver",
          "id": "DEF456",
          "make": "Toyota",
          "model": "Camry",
          "state": "OR",
        },
        Object {
          "color": "blue",
          "id": "GHI789",
          "make": "Hyundai",
          "model": "Elantra",
          "state": "WA",
        },
      ]
    }
    totalCars={4}
  />
</Fragment>
`;
